File: examples/python/src/greet_test.py
"""Unit tests for the greet module."""

import unittest

from greet import greet


class GreetTest(unittest.TestCase):
    def test_greet_world(self) -> None:
        self.assertEqual(greet("World"), "Hello, World!")

    def test_greet_horcrux(self) -> None:
        self.assertEqual(greet("Horcrux"), "Hello, Horcrux!")

    def test_greet_empty(self) -> None:
        self.assertEqual(greet(""), "Hello, !")


if __name__ == "__main__":
    unittest.main()

File: examples/rust/src/greet_test.rs
// examples/rust/src/greet_test.rs
// Integration test for the greet library.

mod greet;

#[cfg(test)]
mod tests {
    use super::greet;

    #[test]
    fn test_greet_world() {
        assert_eq!(greet::greet("World"), "Hello, World!");
    }

    #[test]
    fn test_greet_empty() {
        assert_eq!(greet::greet(""), "Hello, !");
    }
}
